### PR TITLE
Restructure source and resource files 

### PR DESCRIPTION
**Pull request by DyXel**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,27 +41,28 @@ if(WIN32)
...
+	"src/windows/guildsettings/*.cpp"
+	"src/windows/profile/*.hpp"
+	"src/windows/profile/*.cpp"
+	"src/dialogs/*.hpp"
+	"src/dialogs/*.cpp"
 )
 
 add_executable(abaddon ${ABADDON_SOURCES})
+target_include_directories(abaddon PUBLIC ${PROJECT_SOURCE_DIR}/src)
 target_include_directories(abaddon PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (DyXel) [inline CMakeLists.txt:66]: Maybe unnecessary now?

**Comment by NexAdn** [comment]
What is the reason/motivation for this change?

Also, since `etc` in the Unix world is generally associated with begin a config file directory, I'd suggest using a different name. `res` is already a good choice, so I suggest to move all resource files below that directory and move the files which are currently in `res` to a newly created subdirectory of `res` (e.g. `img`).

**Comment by DyXel** [comment]
> What is the reason/motivation for this change?

Clear separation between source files and resources. Makes certain things like GREPping the source code easier and such.

> Also, since etc in the Unix world is generally associated with begin a config file directory, I'd suggest using a different name. 
res is already a good choice, so I suggest to move all resource files below that directory and move the files which are currently in res to a newly created sub directory of res (e.g. img).

At the moment this PR is on hold as @/ouwou is working on merging other branches into master, maybe this is a good time to decide in the final structure for the non-source code parts of the software. I agree with your suggestion and wouldn't mind changing the folder's name so long the separation is clear from a development and distribution standpoint.

**Comment by NexAdn** [comment]
> At the moment this PR is on hold [...]
Well, that isn't very clear, since it's neither marked WIP/Draft in the title nor the PR description.
I assumed this was already like you wanted it to be.

> so long the separation is clear from a development and distribution standpoint.
From a development standpoint: I usually see all the resource files below a directory like `res`. Java even has `src/main/res` as a directory specifically designed for resource files.

From a distro maintainer's PoV: As long as it can be installed below `/usr/share/[packagename]`, there shouldn't be a problem. But I've already managed to package abaddon for Gentoo, so this isn't a problem now and after this PR it won't be a problem either. Moving everything to `res` doesn't really change a thing.

**Comment by ouwou** [comment]
i do like the idea of restructuring it like this. using `res` and `img` as names makes sense too. maybe have this closed until i finalize a couple other things?

**Comment by NexAdn** [comment]
> maybe have this closed until i finalize a couple other things?

You could also mark it as draft so other people see that it is already being worked on. Plus you don't forget that it exists

**Pull request merged**